tests/test_set_state.py

```diff
@@ -1,45 +1,166 @@
 """Test endpoint config option `set_state` and `get_state`."""
 import pytest
+import orjson as json
 
-from coco.test import coco_runner
+from coco.test import coco_runner, endpoint_farm
+from coco.state import State
 
 SET_ENDPT_NAME = "set"
 SET_ENDPT_NAME_INT = "set_int"
+SET_ENDPT_NAME_DICT = "set_dict"
 GET_ENDPT_NAME = "get"
-CONFIG = {"log_level": "INFO", "groups": {"no_group": ["no_host", "doesnt_exist"]}}
+CHECK_HASH_ENDPT_NAME = "check_hash"
+CHECK_HASH_ENDPT_NAME2 = "check_hash2"
+CHECK_STATE_ENDPT_NAME = "check_state"
+CHECK_STATE_ENDPT_NAME2 = "check_state2"
+HASH_ENDPT_NAME = "hash"
+CONFIG = {"log_level": "DEBUG", "groups": {"no_group": ["no_host", "doesnt_exist"]}}
 INT_VAL = 5
 STATE_PATH = "test_state"
 ENDPOINTS = {
     SET_ENDPT_NAME: {"call": {"forward": None}, "set_state": {STATE_PATH: True}},
     SET_ENDPT_NAME_INT: {"call": {"forward": None}, "set_state": {STATE_PATH: INT_VAL}},
+    SET_ENDPT_NAME_DICT: {
+        "call": {"forward": None},
+        "set_state": {STATE_PATH: {"s": {"n": {"a": "fu"}}}},
+    },
     GET_ENDPT_NAME: {"call": {"forward": None}, "get_state": STATE_PATH},
+    CHECK_HASH_ENDPT_NAME: {
+        "group": "test",
+        "values": {"data": "str"},
+        "call": {"forward": {"name": HASH_ENDPT_NAME, "reply": {"state_hash": {"data": "/"}}}},
+    },
+    CHECK_HASH_ENDPT_NAME2: {
+        "group": "test",
+        "values": {"data": "str"},
+        "call": {
+            "forward": {
+                "name": HASH_ENDPT_NAME,
+                "reply": {"state_hash": {"data": "test_state/s/n"}},
+            }
+        },
+    },
+    CHECK_STATE_ENDPT_NAME: {
+        "group": "test",
+        "values": {"data": "str"},
+        "call": {
+            "forward": {"name": HASH_ENDPT_NAME, "reply": {"state": {"data": "test_state/s/n/a"}}}
+        },
+    },
+    CHECK_STATE_ENDPT_NAME2: {
+        "group": "test",
+        "values": {"n": "dict"},
+        "call": {"forward": {"name": HASH_ENDPT_NAME, "reply": {"state": "test_state/s/"}}},
+    },
 }
 
 
+def callback(data):
+    """Reply with the incoming json request."""
+    return data
+
+
+N_HOSTS = 1
+CALLBACKS = {HASH_ENDPT_NAME: callback}
+
+
 @pytest.fixture
-def runner():
+def farm():
+    """Create an endpoint test farm."""
+    return endpoint_farm.Farm(N_HOSTS, CALLBACKS)
+
+
+@pytest.fixture
+def runner(farm):
     """Create a coco runner."""
+    CONFIG["groups"] = {"test": farm.hosts}
     return coco_runner.Runner(CONFIG, ENDPOINTS)
 
 
 def test_get_state(runner):
     """Test get/set_state."""
-    # State should be empty now
-    response = runner.client(GET_ENDPT_NAME)
-    assert "state" in response
-    assert STATE_PATH in response["state"]
-    assert response["state"][STATE_PATH] == {}
```

Comment:
was getting an empty state removed?

Reply:
The state is not empty anymore at this point. The constructors in `check.py` create the parts of the state configured in the configs.